homeassistant/components/somfy/__init__.py

```diff
@@ -0,0 +1,165 @@
+"""
+Support for Somfy hubs.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/somfy/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import callback
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+API = 'api'
+
+DEVICES = 'devices'
+
+REQUIREMENTS = ['pymfy==0.4.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+DOMAIN = 'somfy'
+
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+
+NOTIFICATION_CB_ID = 'somfy_cb_notification'
+NOTIFICATION_OK_ID = 'somfy_ok_notification'
+NOTIFICATION_TITLE = 'Somfy Setup'
+
+ATTR_ACCESS_TOKEN = 'access_token'
+ATTR_REFRESH_TOKEN = 'refresh_token'
+ATTR_CLIENT_ID = 'client_id'
+ATTR_CLIENT_SECRET = 'client_secret'
+
+SOMFY_AUTH_CALLBACK_PATH = '/auth/somfy/callback'
+SOMFY_AUTH_START = '/auth/somfy'
+
+DEFAULT_CACHE_PATH = '.somfy'
```

Comment:
Do we have a convention for the file use as cache, like the name or the path?

Reply:
is it really necessary strictly speaking to have a cache ? What is the purpose of the cache?

Reply:
If it's just a token, you can store it as an attribute of the component, or even better use a config_entry. 

We try to avoid creating extra files at all cost.

Reply:
I also agree.  About  the config_entry, it seems to be use only for the platform. The token has to be unique and is retrieve only once, during component setup. So it seems I can only use attribute. Have you an example? Is the date store in the attribute persistent?

Reply:
Indeed, Nest has a similar workflow except oauth code has to be set by the end user. In my case, the oauth callback redirect  to the HASS instance. At the end, the token is stored in `.storage/core.config_entries`. I will try to understand the config_entry flow. But looking at Nest code, it seems really complex :S

Reply:
I will follow the [documentation](https://developers.home-assistant.io/docs/en/config_entries_config_flow_handler.html). If I understood correctly, I have to create an entry storing only the token and the expiration date, exactly like the Nest component.

Reply:
Not sure you can do that :frowning_face: